### DISCUSSION TITLE
Using let with attributes

### DISCUSSION THREAD

**Discussion by PierreR** [cuelang/cue#938] [Q&A] [answered]
It looks like `let` cannot be used with attributes ? 
```
let Version = *"0.0.0" | string @tag(version)

version: "\(Version)"
```
would fail with the following error (using https://tip.cuelang.org/play/?id=kTgiOQPmiqe):
```
failed to load: missing ',' in struct literal (and 1 more errors)
```
The following works:
```
_version = *"0.0.0"  | string @tag(version)

version: "\(_version)"
```
but I would like to restrain the scope of `_version` to local (ensuring it is  not set somewhere else). In that regard, a `let` seems more appropriate.

Alias would kind of work:
```
Version= "dummy" : *"0.0.0" | string @tag(version)

version: "\(Version)"
```
but does not seem to what I want because the "dummy" label is useless and if re-use somewhere else possibly damaging.

Thanks for your help.

**Comment by PierreR** [comment]
The usecase might still be a valid one but as far as I am concerned this is not blocking or a big deal.
At the end, I have defined what I expect from a version down the file hierarchy in a common place as a definition.

**Comment by mpvl** [comment, marked as answer]
Attributes are indeed not allowed on `let`. I would be quite a substantial change to support, as `let` is not something that is visible in the API, for instance, and which is the only place were attributes are used.

Am I guessing correctly that you are envisioning picking up the attribute in the cue AST rather than the API?

**Reply by PierreR** [reply]
Yes, I don't expect using the API soon (just plain CUE for now).
Thanks for the feedback.

**Reply by mpvl** [reply]
I see your point in how it could be useful, though. Feel free to open a feature request. This will allow tracking if this is a more widely desired feature. Haven’t thought about the implications yet, but barring any issues I could see this work for the tag use case, for example.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/938.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
